app/controllers/api/connect/v3/subscriptions/systems_controller.rb

```diff
@@ -1,4 +1,5 @@
 class Api::Connect::V3::Subscriptions::SystemsController < Api::Connect::BaseController
+  before_action :authenticate_with_token
```

Comment:
This is currently making some tests to fail, but at the same time this is in place in the equivalent Glue code. Moreover, it makes sense to have it since when registering for the first time a system it should also be checked. That being said, if (comparing with other endpoints) this wasn't there, was it because some specific logic that I'm missing? Or this should really be put in here and I need to fix those tests?

Reply:
You are right we could check the request here as well but we would need to make it more smart than it is now.

Right now it checks for all systems if they are authenticated with system credentials, making all systems which are registered without a regcode (valid use case) not working anymore. You can also not register a system without a regcode anymore.

If we want to have this checked I propose we should check if the system is registered with a regcode and only then validate the authentication. But I would discuss the benefits first and move this to its own card if needed.

Reply:
Indeed, we still want to allow registration without a regcode.